tools/wasi/wasi.cpp

```diff
@@ -99,7 +99,7 @@ ExecutionResult environ_sizes_get(Instance& instance, const Value* args, int)
 }
 }  // namespace
 
-bool run(int argc, const char** argv)
+bool run(int argc, const char** argv, std::ostream& err)
```

Comment:
I know you say to make this noexcept, but if not noexcept we could consider just outputting error text as part of an exception? Or we want to pass stdout/stdin here too?

Reply:
This is to be able to inspect the error messages, one step from using `std::cerr` implicitly inside. Further modifications are possible, but I'm not sure what's best.